buildtools/leiningen/leiningen.go

```diff
@@ -0,0 +1,224 @@
+package leiningen
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+// Output is intended to supply a dependency graph from lein output.
+type Output interface {
+	DependencyGraph(string) (graph.Deps, *errors.Error)
+}
+
+// Dependency models a leiningen dependency.
+type Dependency struct {
+	GroupID    string `xml:"groupId"`
+	ArtifactID string `xml:"artifactId"`
+	Version    string `xml:"version"`
+}
+
+// id returns a maven locator from dependency information.
+func (d Dependency) id() string {
+	return d.GroupID + ":" + d.ArtifactID
+}
+
+// Shell implements Output and uses Cmd to generate the DependencyGraph.
+type Shell struct {
```

Comment:
This feels like something that would be useful elsewhere, along with `ShellOutput`, to make behavior more uniform across analyzers